tests/test_pay.py

```diff
@@ -3261,6 +3261,77 @@ def test_mpp_presplit_routehint_conflict(node_factory, bitcoind):
     l1.rpc.pay(inv)
 
 
+def test_delpay_argument_invalid(node_factory, bitcoind):
+    """
+    This test includes all possible combination of input error inside the
+    delpay command.
+    """
+
+    l1, l2 = node_factory.get_nodes(2)
+
+    with pytest.raises(RpcError):
+        l2.rpc.delpay()
+
+    # invoice unpaid
+    inv = l1.rpc.invoice(10 ** 5, 'inv', 'inv')
+    payment_hash = inv["payment_hash"]
+    with pytest.raises(RpcError):
+        l2.rpc.delpay(payment_hash)
+
+    # payment unpaid with wrong status (pending status is a illegal input)
+    with pytest.raises(RpcError):
+        l2.rpc.delpay(payment_hash, 'pending')
+
+    with pytest.raises(RpcError):
+        l2.rpc.delpay(payment_hash, 'invalid_status')
+
+    l2.rpc.connect(l1.info['id'], 'localhost', l1.port)
+    l2.fund_channel(l1, 10 ** 6)
```

Comment:
You could have used `l1, l2 = node_factory.line_graph(2)` above since so far the channel has not been used. That'd save a couple of test-lines

Reply:
In these cases, I could be used the following line of code to have the channels ready to work `node_factory.line_graph(2, fundamount=10**5, wait_for_announce=True)` , right?